lib/appraisal/matrix/rubygems_helper.rb

```diff
@@ -5,20 +5,39 @@
 
 module Appraisal
   module Matrix
-    module RubygemsHelper
-      def versions_to_test(gem_name, minimum_version)
-        # Generate a set to store the versions to test against 
-        versions_to_test = Set.new
-
-        # Load versions from rubygems api
-        URI.parse("https://rubygems.org/api/v1/versions/#{gem_name}.json").open do |raw_version_data|
-          JSON.parse(raw_version_data.read).each do |version_data|
-            version = Gem::Version.new(version_data['number'])
-            versions_to_test << version.segments[0..1].join('.') if version >= minimum_version && !version.prerelease?
+    class RubygemsHelper
+      class << self
+        def versions_to_test(gem_name, version_restrictions, step)
+          # Generate a set to store the versions to test against 
+          versions_to_test = Set.new
+
+          # Load versions from rubygems api
+          URI.parse("https://rubygems.org/api/v1/versions/#{gem_name}.json").open do |raw_version_data|
+            JSON.parse(raw_version_data.read).each do |version_data|
+              version = Gem::Version.new(version_data['number'])
+              versions_to_test << version_for_step(version, step) if include_version?(version, version_restrictions)
+            end
           end
+
+          versions_to_test
         end
 
-        versions_to_test
+        private
+
+        SEGMENT_STEP_SIZES = {
+          major: 1,
+          minor: 2,
+          patch: 3
+        }.freeze
+
+        def version_for_step(version, step)
+          size_for_step = SEGMENT_STEP_SIZES[step] or raise ArgumentError, "unsupported requested version step: #{step}, expected #{SEGMENT_STEP_SIZES.keys}"
+          version.segments.first(size_for_step).join(".")
+        end
+
+        def include_version?(version, version_restrictions)
+          !version.prerelease? && version_restrictions.all? { |dependency| dependency.match?('', version) }
```

Comment:
This code feels weird to me that we're matching on `'', version`. This is specifically coupled with the `VersionArray` class for this to work. I'm curious if there is a do better here to change the arguments around or use a different method 🤔 ...